src/fastq/call_consensus_reads/pipeline.rs

```diff
@@ -100,6 +106,33 @@ impl FASTQStorage {
     }
 }
 
+pub struct OverlappingConsensus {
+    record: Record,
+    likelihood: LogProb,
+}
+
+impl OverlappingConsensus {
+    pub fn new(record: Record, likelihood: LogProb) -> Self {
+        OverlappingConsensus { record, likelihood }
+    }
+}
+
+pub struct NonOverlappingConsensus {
+    f_record: Record,
+    r_record: Record,
+    likelihood: LogProb,
+}
+
+impl NonOverlappingConsensus {
+    pub fn new(f_record: Record, r_record: Record, likelihood: LogProb) -> Self {
```

Comment:
You can remove the constructors, and directly initialize the structs. We anyway use them only internally.